src/DirectoryManager.cpp

```diff
@@ -0,0 +1,59 @@
+#include "DirectoryManager.h"
+#include <iostream>
+#include <sstream>
+#include <unistd.h>
+#include <sys/wait.h>
+
+
+/*
+* get the current directory.
+* @return the current directory as string.
+*/
+std::string DirectoryManager::getCurrentDirectory()
+{
+    char cwd[PATH_MAX];
+    return (getcwd(cwd, sizeof(cwd)) != nullptr) ?
```

Comment:
This way of getting cwd is a bit old fashion, check this out (a more modern way): https://en.cppreference.com/w/cpp/filesystem/current_path